File: app/src/main/java/com/gmail/fattazzo/formula1world/domain/F1Race.java
package com.gmail.fattazzo.formula1world.domain;

import java.io.Serializable;

/**
 * @author fattazzo
 *         <p/>
 *         date: 06/06/17
 */
public class F1Race implements Serializable {

    public int year;

    public int round;

    public String url;

    public String name;

    public F1Circuit circuit;

    public String date;

    public String time;

    public F1Race(int year, int round, String url, String name, F1Circuit circuit, String date, String time) {
        this.year = year;
        this.round = round;
        this.url = url;
        this.name = name;
        this.circuit = circuit;
        this.date = date;
        this.time = time;
    }

    @Override
    public String toString() {
        return "F1Race{" +
                "year=" + year +
                ", round=" + round +
                ", url='" + url + '\'' +
                ", name='" + name + '\'' +
                ", circuit=" + circuit +
                ", date='" + date + '\'' +
                ", time='" + time + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        F1Race f1Race = (F1Race) o;

        if (year != f1Race.year) return false;
        return round == f1Race.round;

    }

    @Override
    public int hashCode() {
        int result = year;
        result = 31 * result + round;
        return result;
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/domain/F1Race.kt
/*
 * Project: total-gp-world
 * File: F1Race.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.domain

import java.io.Serializable

/**
 * @author fattazzo
 *
 *
 * date: 06/06/17
 */
class F1Race(var year: Int, var round: Int, var url: String, var name: String, var circuit: F1Circuit, var date: String?, var time: String?) : Serializable {

    override fun toString(): String {
        return "F1Race{" +
                "year=" + year +
                ", round=" + round +
                ", url='" + url + '\'' +
                ", name='" + name + '\'' +
                ", circuit=" + circuit +
                ", date='" + date + '\'' +
                ", time='" + time + '\'' +
                '}'
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false

        val f1Race = o as F1Race?

        return if (year != f1Race!!.year) false else round == f1Race.round

    }

    override fun hashCode(): Int {
        var result = year
        result = 31 * result + round
        return result
    }
}
